extensions/wikia/ArticleVideo/ArticleVideoController.class.php

```diff
@@ -12,9 +12,15 @@ public function featured() {
 
 			$this->setVal( 'videoDetails', $featuredVideoData );
 			if ( ArticleVideoContext::isJWPlayer( $featuredVideoData ) ) {
-                $this->setVal( 'jwplayerInstance', file_get_contents($this->getApp()->wg->extensionsPath . '/wikia/ArticleVideo/scripts/featured-video.jwplayer.instance.js') );
-				$this->setVal( 'featuredVideoAutoplay', file_get_contents($this->getApp()->wg->extensionsPath . '/wikia/ArticleVideo/scripts/featured-video.autoplay.js') );
-				$this->setVal( 'jwplayerScript', file_get_contents($this->getApp()->wg->extensionsPath . '/wikia/ArticleVideo/scripts/featured-video.jwplayer.instant.js') );
+				$jwPlayerScript = RequestContext::getMain()->getOutput()->getResourceLoader()->getModule( 'ext.ArticleVideo.jw' )->getScript(
```

Comment:
can we cache `ReqContext.getMain`? it's used twice

Reply:
In a Nirvana context you can also use `$this->getContext()`

Reply:
Removed this call and used $this->getContext() on the context we use at the beg of the method, thanks